rust/chains/hyperlane-ethereum/src/tx.rs

```diff
@@ -92,14 +96,18 @@ where
     M: Middleware + 'static,
     D: Detokenize,
 {
+    // either use the pre-estimated gas limit or estimate it
+    let estimated_gas_limit: U256 = match tx.tx.gas() {
```

Comment:
We don't set the ContractCall's gas prior to `process_contract_call` calling this function

https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/f800297259a0932cfbce9e988648efc9254848fd/rust/chains/hyperlane-ethereum/src/contracts/mailbox.rs#L301-L312

So we end up hitting the None branch and estimating gas again for each process call in a batch or a non-batched message when it's submitted, and only hit the Some branch for a batch tx. Is this intentional?

I guess not the end of the world but still feels a bit wasteful / confusing. Maybe `process_contract_call` can set the gas if `tx_gas_estimate` is Some? This used to be the intention of `add_gas_overrides` but now that the behavior has changed a bit, this setting it in `process_contract_call` may make sense

Reply:
Ended up doing the batch tx gas estimation in the `batch` constructor function, seems cleaner. So now I'm doing both the estimate for the batch and the summing when building the batch `ContractCall`, and I no longer need to abuse the semantics of gas overrides. 

I did keep the logic to pick the `max` between the override and the gas estimate though, to avoid override misconfigs preventing us from submitting.

So now the submission flow of a batch will set the gas limit based on `max(override, max(individual_op_sum, batch_estimate))`, whereas the flow of a single message per tx will just be `max(override, op_estimate)`